src/statistics_tracking/GenericStatistic.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Newtonsoft.Json;
+
+public abstract class GenericStatistic<T> : IStatistic
+{
+    public GenericStatistic(T value, StatsTrackerEvent linkedEvent)
+    {
+        Value = value;
+        LinkedEvent = linkedEvent;
+    }
+
+    public StatsTrackerEvent LinkedEvent { get; set; }
+
+    public T Value { get; protected set; }
+
+    public abstract void Increment(T value);
+}
+
+public class SimpleStatistic : GenericStatistic<int>
+{
+    public SimpleStatistic(StatsTrackerEvent @event)
+        : base(0, @event)
```

Comment:
I prefer having `:` on the previous line, I think that looks better to me and its the style I've been enforcing everywhere in THrive so this is now inconsistent style to fit with all of the rest of the code.